src/main/java/com/notably/model/suggestion/SuggestionModelImpl.java

```diff
@@ -0,0 +1,63 @@
+package com.notably.model.suggestion;
+
+import java.util.List;
+import java.util.Optional;
+
+import javafx.beans.property.Property;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+
+/**
+ * The implementation class of SuggestionModel.
+ */
+public class SuggestionModelImpl implements SuggestionModel {
+    private ObservableList<SuggestionItem> suggestions;
+    private String commandInputText;
+    private Property<Optional<String>> commandTextProperty;
+
+    @Override
+    public ObservableList<SuggestionItem> getSuggestions() {
+        return suggestions;
+    }
+
+    @Override
+    public void setSuggestions(List<SuggestionItem> suggestions) {
+        this.suggestions = FXCollections.observableArrayList(suggestions);
+    }
+
+    // TODO: update commands
+    @Override
+    public Property<Optional<String>> getCommandTextProperty() {
+        if (commandInputText.startsWith("open")) {
+            commandTextProperty.setValue(Optional.of("Open a note"));
+        } else if (commandInputText.startsWith("search")) {
+            commandTextProperty.setValue(Optional.of("Search a note"));
+        } else if (commandInputText.isEmpty()) {
+            commandTextProperty.setValue(Optional.empty());
+        } else {
+            commandTextProperty.setValue(Optional.of("TODO"));
+        }
+
+        return commandTextProperty;
+    }
```

Comment:
Hmm is not the way we work with observables. Instead, you should straight away return the property entity. That is, do `return commandTextProperty` only.

Reply:
https://docs.oracle.com/javafx/2/binding/jfxpub-binding.htm Please read this up!